### ISSUE TITLE
Support for selecting (uploading if needed) application version in Elastic Beanstalk

### ISSUE THREAD

**Issue by zette** [hashicorp/terraform#6681]
Hi,
When creating environment in new Application from Elastic Benastalk resource, terraform does not provide a way to select or upload new version of application to AWS. 
This leads to errors like during terraform apply:

```
Error applying plan:

1 error(s) occurred:

* aws_elastic_beanstalk_environment.EBSENV: InvalidParameterValue: No application version exists to deploy for Application named Parking.
    status code: 400, request id: b1c39551-1b4f-11e6-bc0d-ffe15606ffdc
```

It seems that the default AWS behavior stated in documentation, that when no application version is specified, then "Sample" one is deployed does not work always.
It would be great that it would be possible for terraform to specify and optionally upload new application version during environment creation. Local file/S3/git sources for application would be even better if possible.

**Comment by karim-h** [comment]
Just to add to Zette's comments:
This Elastic Beanstalk resource is not really useful unless there is a way to deploy an application to the stack that has been created.

**Comment by karim-h** [comment]
Update: functionality has been [added](https://github.com/hashicorp/terraform/pull/5770) but the docs have not been updated to reflect it.

**Comment by kcrawley** [comment]
@karim-h i've run into the same issue as you. there are open PRs pending approval which address this problem. i've compiled 7407 and had success with uploading an initial version which puts the stack in a usable state.

https://github.com/hashicorp/terraform/pull/7407
https://github.com/hashicorp/terraform/pull/5770

**Comment by grebois** [comment]
+1

**Comment by marchub** [comment]
+1

**Comment by ekarlsso** [comment]
I'm having a problem creating an elasticbeanstalk application for Node. Fails with the "No application version exists to deploy for Application". So would be good to have this possibility available. At least for the initial setup.

**Comment by grebois** [comment]
+1

**Comment by nungster** [comment]
+1

**Comment by matskrengel** [comment]
+1

**Comment by joeharrison714** [comment]
+1

**Comment by joeharrison714** [comment]
+1

**Comment by dev1x** [comment]
+1

**Comment by pasviegas** [comment]
+1

**Comment by jweber-e2w** [comment]
+1

**Comment by marioharper** [comment]
+1

**Comment by dharrisio** [comment]
@catsby / @stack72 This functionality has been added with #5770, so this should be good to close.

**Comment by stack72** [comment]
Closed via #5770
